extensions/WSLExtension/DistributionDefinitions/DistributionDefinitionHelper.cs

```diff
@@ -0,0 +1,115 @@
+using WSLExtension.Helpers;
+using YamlDotNet.Serialization;
+using static WSLExtension.Constants;
+
+namespace WSLExtension.DistributionDefinitions;
+
+/// <summary>
+/// Provides definition information about all the WSL distributions that can be found at
+/// <see cref="Constants.KnownDistributionsWebJsonLocation"/>.
+/// </summary>
+public class DistributionDefinitionHelper : IDistributionDefinitionHelper
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(DistributionDefinitionHelper));
+
+    private readonly IHttpClientFactory _httpClientFactory;
+
+    private readonly PackageHelper _packageHelper = new();
+
+    private readonly Architecture _osArchitecture;
+
+    private readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        PropertyNameCaseInsensitive = true,
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+    };
+
+    public DistributionDefinitionHelper(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+        _osArchitecture = RuntimeInformation.OSArchitecture;
+    }
+
+    /// <inheritdoc cref="IDistributionDefinitionHelper.GetDistributionDefinitionsAsync"/>
+    public async Task<Dictionary<string, DistributionDefinition>> GetDistributionDefinitionsAsync()
+    {
+        try
+        {
+            // Get the update to date distribution definitions from WSL GitHub repository.
+            var client = _httpClientFactory.CreateClient();
+            var distributionDefinitionsJson = await client.GetStringAsync(KnownDistributionsWebJsonLocation);
+            var webDefinitions = JsonSerializer.Deserialize<DistributionDefinitions>(distributionDefinitionsJson, _jsonOptions);
+            var distributionDefinitionsMap = new Dictionary<string, DistributionDefinition>();
+
+            foreach (var definition in webDefinitions!.Values)
+            {
+                // Only supported distributions for this machine.
+                if (ShouldAddDistribution(definition))
+                {
+                    distributionDefinitionsMap.Add(definition.Name, definition);
+                }
+            }
+
+            // Merge the local distribution information we have stored in DistributionDefinition.yaml with the one above.
+            var uri = new Uri(KnownDistributionsLocalYamlLocation);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var localYamlDefinitionsFile = await FileIO.ReadTextAsync(storageFile);
+            var localYamlDefinitions = BuildYamlDeserializer().Deserialize<List<DistributionDefinition>>(localYamlDefinitionsFile);
+            foreach (var localYamlDefinition in localYamlDefinitions)
+            {
+                // Ignore distributions that we have in the local yaml file but are no longer present in the web file.
+                if (!distributionDefinitionsMap.TryGetValue(localYamlDefinition.Name, out var definitionFromWeb))

```

Comment:
Why?

Reply:
we use this json file as our single source of truth: https://aka.ms/distributionsinfojson. Wsl.exe also uses this json file to install and register a distribution from the web. If our local yaml file is old (older version of Dev Home), it may have items that no longer exist in the json file. So, there is no need to iterate through them.